rllib/algorithms/impala/impala.py

```diff
@@ -395,9 +378,20 @@ def validate(self) -> None:
                     "term/optimizer! Try setting config.training("
                     "_tf_policy_handles_more_than_one_loss=True)."
                 )
+        # learner hps need to be updated inside of config.validate in order to have
+        # the correct values for when a user starts an experiment from a dict. This is
+        # as oppposed to assigning the values inthe builder functions such as `training`
```

Comment:
I found your explanation from the PR itself more informative. Together they would read:

"LearnerHPs need to be updated inside of `AlgorithmConfig.validate()` in order to work with tune's HP search. This is because, if a user passes their experiment config to tune.run as a dictionary, the LearnerHPs will not be ever set. In that case, downstream Learners will not receive the passed HPs."

Reply:
ill throw it in my current pr. I've never seen my tests this green in my life and want to keep it that way

Reply:
am just gonna say we shouldn't make changes in random PR just because we don't want to touch CI again.
attributions will be wrong, and people will get confused if they git blame.
maybe it's better to send a separate comment PR for it, instead of including it in whatever next PR you are working on.

Reply:
although have to say that this is way too green ... :)